include/seqan3/io/stream/debug_stream.hpp

```diff
@@ -312,7 +324,8 @@ inline debug_stream_type & operator<<(debug_stream_type & s, alphabet_t const l)
 template <std::ranges::InputRange rng_t>
 inline debug_stream_type & operator<<(debug_stream_type & s, rng_t && r)
 //!\cond
-    requires requires (reference_t<remove_cvref_t<rng_t>> l) { { debug_stream << l }; } &&
+    requires !std::Same<remove_cvref_t<reference_t<std::remove_const_t<rng_t>>>, filesystem::path> &&
+             requires (reference_t<remove_cvref_t<rng_t>> l) { { debug_stream << l }; } &&
```

Comment:
This should not be required, the above overload is more specialised.

Reply:
When I tried it without it, it still produced an ICE. 
It still tries to evaluate the requires expression, right?

Reply:
@h-2 As The ICE is produced when evaluating the requires statement in line 328. As @eseiler said, this is a hack to avoid the ICE by taking advantage of the lazy evaluation of requires statements.

Reply:
Ah, what a mess. But then please check instead whether the reference types is equal to the type itself. That's more generic than checking for path directly.

Reply:
I check now whether the reference type is equal to the type.